app/javascript/components/search/results/StudySearchResult.jsx

```diff
@@ -170,17 +175,109 @@ function studyTypeBadge(study) {
   }
 }
 
+/**
+ * Log analytics about search result selection to Mixpanel
+ *
+ * This logging differentiates properties about the selected result from
+ * properties about other aspects of the search -- e.g. the full result set
+ * and query -- by prefixing the latter with a "namespace", i.e. `results:`.
+ *
+ * @param {Object} study Selected result
+ * @param {Object} logProps Other properties to log
+ *
+ * TODO (SCP-4256): Refine logged properties for search result selection
+ * TODO (SCP-4257): Eliminate duplicate default/custom search event logging
+ */
+export function logSelectSearchResult(study, logProps={}) {
+  logProps = Object.assign(
+    {
+      studyAccession: study.accession,
+      context: 'study',
+      scope: 'global'
+    },
+    study,
+    logProps
+  )
+  delete logProps.accession // Redundant with more-conventional studyAccession
+
+  Object.entries(logProps.results).forEach(([key, value]) => {
+    if (['termList', 'geneList'].includes(key)) {
+      // Redundant with pre-existing `terms` and `genes` Mixpanel props
+      return
+    }
+    logProps[`results:${key}`] = value
+  })
+  delete logProps.results // Remove nested property; flattened above
+
+  // Number of searches done before this result was selected.
+  // This is needed to easily analyze changes in click-through rate (CTR).
+  logProps.numSearches = numSearches
+
+  // We don't log study name, like Terra doesn't log workspace name, per
+  // original DSP Mixpanel design doc
+  delete logProps.name
+  delete logProps.study_url // Has name
+  delete logProps.description // Too similar to name
+  delete logProps['results:studies'] // Includes above
+
+  // Objects can't be queried in Mixpanel, so flatten matchByData
+  if (logProps['results:matchByData'] !== null) {
+    Object.entries(logProps['results:matchByData']).forEach(([key, value]) => {
+      logProps[`results:${key}`] = value
+    })
+  }
+  delete logProps['results:matchByData']
+
+  let refinedLogProps = {}
+  Object.entries(logProps).forEach(([key, value]) => {
+    // Counts of "foo" in Mixpanel are conventionally structured as `numFoos`.
+    if (key.slice(-6) === '_count') {
+      const namespace = key.includes('results:') ? 'results:' : ''
+      // Convert e.g. `gene_count` to `numGenes`
+      key = `${namespace}num${key[0].toUpperCase() + key.slice(1).replace('_count', '') }s`
+    }
+    if (key.slice(0, 13) === 'results:total') {
```

Comment:
I understand how `_count` isn't really JS-ified, and so changing those might be worth the effort.  But we called these properties `totalX` and not `numX` for a reason--it distinguished that they were the total available, rather than just the number returned.  I'm not sure what benefit we get by renaming them, and why that's worth the possible confusion.  I guess we could call them `numTotalPages` if it makes navigating mixpanel easier to have all the count-y variables alphabetically close? 
And again, there's not many of them, I'd lean towards writing them out

Reply:
Yup, analysis UX consistency and simplicity are the main benefit of renaming these.  I've gone with `numTotalFoo` in cases of `totalFoo` as you suggest, and now write them out; 8d57e81.